docs/source/app-dev/services.rst

```diff
@@ -94,6 +94,21 @@ The command submission service deduplicates submitted commands based on their :r
 
 For details on how to use command deduplication, see the :ref:`Application Architecture Guide <command-deduplication>`.
 
+.. note:: Ledgers with command deduplication across participants
+
+  - The ledger may extend the deduplication period specified in the request arbitrarily, up until the maximum deduplication duration specified in the :ref:`ledger configuration <ledger-configuration-service>`.
```

Comment:
```suggestion
  - The ledger may extend the deduplication period specified in the request arbitrarily, even beyond the maximum deduplication duration specified in the :ref:`ledger configuration <ledger-configuration-service>`.
```

Reply:
Are you suggesting 'even beyond' because of accounting for time skew or for a different reason?

Reply:
Time skews are one thing where it can already happen for kv-based ledgers.
The other thing is that Canton has to report a deduplication offset rather than a duration; but if the submission had specified a duration, then there may not be a suitable completion offset around the time where the deduplication duration starts. So if the deduplication duration already is max dedup time, we cannot really return an offset that was published within the max dedup duration without incurring the risk that we're shortening the effective deduplication duration. So instead, Canton doesn't even bother to find a completion offset close the the beginning of the deduplication duration. Instead, we return the first known offset on the ledger or the offset after the last completion for the change ID. So for an unpruned ledger and a fresh change ID, the effective deduplication period is unbounded. Given that longer durations provide a stronger statement to the application, the application should actually be happy about this.

Reply:
If there is already a completion for the same change ID, then Canton typically does not extend the deduplication period before the completion if the earlier completion wouldn't normally fall into the deduplication period. But as always, with clock drift, this may appear to the application as if some extension happened.